sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/NumberPatternShouldBeRegular.cs

```diff
@@ -150,5 +129,24 @@ private static string StripNumericPreAndSuffix(string numericToken)
 
             return numericToken;
         }
+
+        private static bool HasIrregularGroupLengths(int[] groupLengths)
+        {
+            if (groupLengths.Length < 2)
+            {
+                return false;
+            }
+
+            // the second group is assumed to be leading.
```

Comment:
I don't understand what "leading" means?

```suggestion
            // the first group is allowed to contain less digits that the other ones, so take the "regular" length from the second group
```